src/core/data/store.inl

```diff
@@ -269,12 +269,39 @@ void UnboundRegionField::bind_data(Buffer<T, DIM>& buffer, const Point<DIM>& ext
   bound_ = true;
 }
 
-template <typename T, int DIM>
+template <typename T>
+void Store::check_accessor_type() const
+{
+  auto in_type = legate_type_code_of<T>;
+  // Test exact match for primitive types
+  if (in_type != this->code() && in_type != Type::Code::INVALID) {
+    log_legate.error(
+      "Type mismatch: %s accessor to a %s store. Disable type checking via "
+      "accessor template parameter if this is intended.",
+      typeid(T).name(),
```

Comment:
@RAMitchell have you actually tried to exercise this branch? I'd make sure `typeid` prints out the right type name. (I believe it should, but I also vaguely remember `typeid(T).name()` returned a weird string when `T` was one of the types defined in stdint.h.)

Reply:
I've been using this branch for a couple weeks successfully. The name is implementation defined, you get something like 'f' for float on gcc. Let me see if I can convert it to a legate type then print it for consistency.